main.go

```diff
@@ -321,11 +332,46 @@ func awss3(w http.ResponseWriter, r *http.Request) {
 	setStrHeader(w, "ETag", obj.ETag)
 	setTimeHeader(w, "Last-Modified", obj.LastModified)
 
-	if obj.ContentRange != nil && len(*obj.ContentRange) > 0 {
-		w.WriteHeader(http.StatusPartialContent)
+	httpStatus := determineHttpStatus(obj)
+
+	w.WriteHeader(httpStatus)
+}
+
+func determineHttpStatus(obj *s3.GetObjectOutput) int {
```

Comment:
__[Golint]__ <sub>posted by Sider</sub>

func determineHttpStatus should be determineHTTPStatus